File: infrastructure/PrEP/skills/code-vuln-analysis/scripts/fixtures/vuln-node-app/src/safe_query.js
// safe_query.js - Safe user lookup endpoint
// Uses parameterized queries to prevent SQL injection
const express = require('express');
const router = express.Router();
const mysql = require('mysql');

const db = mysql.createConnection({
  host: 'localhost', user: 'app', password: 'app_pass', database: 'myapp'
});

router.get('/safe-user', (req, res) => {
  db.query("SELECT * FROM users WHERE name = ?", [req.query.name], (err, rows) => {
    if (err) return res.status(500).json({ error: 'query failed' });
    res.json({ users: rows });
  });
});

module.exports = router;

File: infrastructure/PrEP/skills/code-vuln-analysis/scripts/fixtures/vuln-python-app/src/safe_query.py
"""safe_query.py - Secure user lookup endpoint.

Demonstrates safe SQL with parameterized queries.
"""
import sqlite3
from flask import Blueprint, request, jsonify
bp = Blueprint('safe_users', __name__)

@bp.route('/safe-user')
def safe_get_user():
    conn = sqlite3.connect('app.db')
    cursor = conn.cursor()
    name = request.args.get('name', '')
    cursor.execute("SELECT * FROM users WHERE name = %s", (name,))
    user = cursor.fetchone()
    conn.close()
    return jsonify({"user": user} if user else {"error": "not found"})
